pkg/operator/configobservation/images/observe_images.go

```diff
@@ -51,8 +55,134 @@ func ObserveInternalRegistryHostname(genericListers configobserver.Listers, reco
+	// has special significance.
+	externalRegistryHostnames := configImage.Spec.ExternalRegistryHostnames
+	externalRegistryHostnames = append(externalRegistryHostnames, configImage.Status.ExternalRegistryHostnames...)
+
+	if len(externalRegistryHostnames) > 0 {
+		hostnames, err := convert(externalRegistryHostnames)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+		err = unstructured.SetNestedField(observedConfig, hostnames, externalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
 		}
 	}
+
+	if !reflect.DeepEqual(o, externalRegistryHostnames) {
+		recorder.Eventf("ObserveExternalRegistryHostnameChanged", "External registry hostname changed to %v", externalRegistryHostnames)
+	}
+
 	return observedConfig, errs
 }
+
+// ObserveAllowedRegistriesForImport maps the user provided list of allowed registries for importing images to the kube api server
+// configuration.
+func ObserveAllowedRegistriesForImport(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	allowedRegistriesForImportPath := []string{"imagePolicyConfig", "allowedRegistriesForImport"}
+	o, _, _ := unstructured.NestedSlice(existingConfig, allowedRegistriesForImportPath...)
+	if len(o) > 0 {
+		err := unstructured.SetNestedSlice(prevObservedConfig, o, allowedRegistriesForImportPath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	if !listers.ImageConfigSynced() {
+		glog.Warning("images.config.openshift.io not synced")
+		return prevObservedConfig, errs
+	}
+
+	// now gather the cluster config and turn it into the observed config
+	observedConfig := map[string]interface{}{}
+	configImage, err := listers.ImageConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("image.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+
+	if len(configImage.Spec.AllowedRegistriesForImport) > 0 {
+		allowed, err := convert(configImage.Spec.AllowedRegistriesForImport)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+		err = unstructured.SetNestedField(observedConfig, allowed, allowedRegistriesForImportPath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	if !reflect.DeepEqual(o, configImage.Spec.AllowedRegistriesForImport) {
+		recorder.Eventf("ObserveAllowedRegistriesForImport", "Allowed registries for import changed to %v", configImage.Spec.AllowedRegistriesForImport)
+	}
+
+	return observedConfig, errs
+}
+
+func convert(o interface{}) (interface{}, error) {
```

Comment:
godoc.

What does this actually do? Convert to JSON? It needs a better name.

Reply:
i'll godoc it but i'm not interested in a naming debate. it's a private function in code my team (and only my team) owns.